include/action.h

```diff
@@ -2,19 +2,48 @@
 #ifndef __LABWC_ACTION_H
 #define __LABWC_ACTION_H
 
-struct server;
+/* Forward declare arguments */
 struct view;
+struct server;
+struct wl_list;
+struct wlr_box;
 
 struct action {
-	uint32_t type;
-	char *arg;
-	struct wl_list link;
+	struct wl_list link;  /* struct keybinding.actions,
+	                       * struct mousebinding.actions,
+	                       * struct menuitem.actions */
+
+	uint32_t type;        /* enum action_type */
```

Comment:
Wouldn't it be better to declare the enum in this header-file and just make this `enum action_type type`.
I'm thinking that other translation units will want to do things like `if (action->type == LAB_ACTION_WHATEVER)`

Reply:
I don't know if we actually need that kind of introspection outside of `action.c`.
The idea was that nobody else should care about the internals of `action.c`. The outside API is either string based (`action_create()`) or `struct action *` based so the enum isn't ever used anywhere other than in `action.c` itself.

The action names are internally translated into the enum in `action_create()` and `actions_run()` gets a *wl_list of action structs as argument.
Maybe we could start using the same notation as wlroots in just putting  a `/* private */` on top of struct members like this?